glib/src/subclass/shared.rs

```diff
@@ -0,0 +1,229 @@
+// Take a look at the license at the top of the repository in the LICENSE file.
+
+//! Module for registering shared types for Rust types.
+
+use crate::translate::*;
+
+pub trait RefCounted: Clone + Sized + 'static {
```

Comment:
Hmm this trait should actually be `unsafe`. You can do an implementation that compiles fine but will fail at runtime, e.g. by making `ref_()` simply pas through the pointer.

```suggestion
pub unsafe trait RefCounted: Clone + Sized + 'static {
```

Reply:
Note: `unsafe fn`s in traits are unsafe to *call*, `unsafe trait`s are unsafe to implement.